### ISSUE TITLE
Questions about the training dataset

### ISSUE THREAD

**Issue by LeoDuhz** [Murkey8895/foldsformer#1]
Hi,
Thanks for your great work which helps me a lot and your code is really well written and easy to understand. However, i have a question about your dataset composition. In your paper(Page 4 Section 3.C) you mention that the training consists of random actions and expert demonstrations. But in your code utils/prepare_data_list.py line 65, it seems that you only split the random action dataset.
I wonder if it is necessary or how to fuse the expert demonstration data and random action data into the training dataset. What's more, in your paper you collect 6000 trajectories of random actions and only 100 trajectories of expert demonstration, i wonder if the number of demonstration data is enough or why this number is 60 times smaller than random actions data.

Thanks in advance for your help!!!

**Comment by Murkey8895** [comment]
Hi LeoDuhz,
Thanks for your interest in my work! 
1) The Foldsformer is first trained on a large dataset of random actions, and then fine-tuned on the small dataset of expert demonstrations. During fine-tuning, the learning rate of encoder and decoders are 1e-5 and 1e-4, respectively. You can also use the "utils/prepare_data_list.py" to preapre the expert demonstrations for training.

2) After trained on a large dataset of random actions, Foldsformer can already successfully perform the tasks when the configuration of the cloth in the observation varies a little from that in the sub-goals. Thus, 100 expert demonstrations are empirically enough for fine-tuing.

**Comment by LeoDuhz** [comment]
@Murkey8895 I understand it. Thanks again for your answer and your great work!!!

**Comment by LeoDuhz** [comment]
Hi bro, i still have a question about the demonstration data used to train and eval the algorithm.

Following your instruction, i generate the square and rectangle configs using `python generate_configs.py --num_cached 100 --cloth_type square
` which are further used to generate the demonstration data and fine-tune the network, as you mentioned above. Then i download your initial evaluation configs [here](https://drive.google.com/drive/folders/1_kZpRBu9bMmt-gFFLxKw7Ih7NOfpbxI-), it seems that those configs used for training and evaluation are quite similar, e.g.: for square size, both `square100.pkl` i generate for training and `square.pkl` you provide for evaluation are all between 50-60. This confuses me now and there might be something that i misunderstand. Could you please help me for this?

Thank a lot for your help!

**Comment by LeoDuhz** [comment]
Hi bro, i still have a question about the demonstration data used to train and eval the algorithm.

Following your instruction, i generate the square and rectangle configs using `python generate_configs.py --num_cached 100 --cloth_type square
` which are further used to generate the demonstration data and fine-tune the network, as you mentioned above. Then i download your initial evaluation configs [here](https://drive.google.com/drive/folders/1_kZpRBu9bMmt-gFFLxKw7Ih7NOfpbxI-), it seems that those configs used for training and evaluation are quite similar, e.g.: for square size, both `square100.pkl` i generate for training and `square.pkl` you provide for evaluation are all between 50-60. This confuses me now and there might be something that i misunderstand. Could you please help me for this?

Thank a lot for your help!

**Comment by Murkey8895** [comment]
There are truly some overlaps of the cloth size between training and evaluation. In the implementation, the cloth size in the demonstrations ranges from 50 to 55, while the cloth size in the evaluation ranges from 50 to 60. We are sorry for not providing these details and related analysis. 
Have a good weekend!

**Comment by LeoDuhz** [comment]
Well i understand it, thank you for illustrating this! Wish you an enjoyable weekend!
